src/Device/AndroidSensors.cpp

```diff
@@ -287,6 +288,36 @@ DeviceDescriptor::OnHeartRateSensor(unsigned bpm) noexcept
   e.Commit();
 }
 
+void
+DeviceDescriptor::OnEngineSensors(Temperature cht,
+                                  Temperature egt,
+                                  Temperature oat,
+                                  AtmosphericPressure pressure,
+                                  uint16_t revs_per_sec) noexcept
+{
+  const auto e = BeginEdit();
+  NMEAInfo &basic = *e;
+  basic.UpdateClock();
+  basic.alive.Update(basic.clock);
+  basic.engine_state.revs_per_sec = revs_per_sec;
+  basic.engine_state.revs_per_sec_available.Update(basic.clock);
+
+  if(pressure.IsPlausible()){
+    basic.static_pressure = pressure;
+    basic.static_pressure_available.Update(basic.clock);
+  }else{
+    basic.static_pressure_available.Clear();
+  }
+
+  basic.engine_state.cht_temperature = cht;
+  basic.engine_state.cht_temperature_available.Update(basic.clock);
+  basic.engine_state.egt_temperature = egt;
+  basic.engine_state.egt_temperature_available.Update(basic.clock);
+  basic.temperature = oat;
+  basic.temperature_available = true;
```

Comment:
So `oat` has really nothing to do with the engine - there's already `SensorListener::OnTemperature()`, and your product should use that instead of adding a redundant parameter to `OnEngineSensors()`.